src/ansys/fluent/core/services/app_utilities.py

```diff
@@ -0,0 +1,252 @@
+        return self._stub.StopPythonJournal(request, metadata=self._metadata)
+
+    def is_beta_enabled(
+        self, request: AppUtilitiesProtoModule.IsBetaEnabledRequest
+    ) -> AppUtilitiesProtoModule.IsBetaEnabledResponse:
+        """Is beta enabled RPC of AppUtilities service."""
+        return self._stub.IsBetaEnabled(request, metadata=self._metadata)
+
+    def is_wildcard(
+        self, request: AppUtilitiesProtoModule.IsWildcardRequest
+    ) -> AppUtilitiesProtoModule.IsWildcardResponse:
+        """Is wildcard RPC of AppUtilities service."""
+        return self._stub.IsWildcard(request, metadata=self._metadata)
+
+    def is_solution_data_available(
+        self, request: AppUtilitiesProtoModule.IsSolutionDataAvailableRequest
+    ) -> AppUtilitiesProtoModule.IsSolutionDataAvailableResponse:
+        """Is solution data available RPC of AppUtilities service."""
+        return self._stub.IsSolutionDataAvailable(request, metadata=self._metadata)
+
+    def register_pause_on_solution_events(
+        self, request: AppUtilitiesProtoModule.RegisterPauseOnSolutionEventsRequest
+    ) -> AppUtilitiesProtoModule.RegisterPauseOnSolutionEventsResponse:
+        """Register on pause solution events RPC of AppUtilities service."""
+        return self._stub.RegisterPauseOnSolutionEvents(
+            request, metadata=self._metadata
+        )
+
+    def resume_on_solution_event(
+        self, request: AppUtilitiesProtoModule.ResumeOnSolutionEventRequest
+    ) -> AppUtilitiesProtoModule.ResumeOnSolutionEventResponse:
+        """Resume on solution event RPC of AppUtilities service."""
+        return self._stub.ResumeOnSolutionEvent(request, metadata=self._metadata)
+
+    def unregister_pause_on_solution_events(
+        self, request: AppUtilitiesProtoModule.UnregisterPauseOnSolutionEventsRequest
+    ) -> AppUtilitiesProtoModule.UnregisterPauseOnSolutionEventsResponse:
+        """Unregister on pause solution events RPC of AppUtilities service."""
+        return self._stub.UnregisterPauseOnSolutionEvents(
+            request, metadata=self._metadata
+        )
+
+    def exit(
+        self, request: AppUtilitiesProtoModule.ExitRequest
+    ) -> AppUtilitiesProtoModule.ExitResponse:
+        """Exit RPC of AppUtilities service."""
+        return self._stub.Exit(request, metadata=self._metadata)
+
+
+class AppUtilities:
+    """AppUtilities."""
+
+    def __init__(self, service: AppUtilitiesService):
+        """__init__ method of AppUtilities class."""
+        self.service = service
+
+    def get_product_version(self, with_patch: bool = True) -> Any:
+        """Get product version."""
+        request = AppUtilitiesProtoModule.GetProductVersionRequest()
+        response = self.service.get_product_version(request)
+        if with_patch:
+            return f"{response.major}.{response.minor}.{response.patch}"
+        else:
+            return f"{response.major}{response.minor}"
```

Comment:
@hpohekar The formatting seems to be wrong here.

Reply:
The attempt to return `FluentVersion` gives circular import issue we have removed `with_patch` and we are returning the string now.

Reply:
`FluentVersion` should be usable without any such issues. There is no intrinsic problem there. Are you provoking this through your imports in this module? I commented earlier on the unnecessary import of `ansys.fluent.core`.

If you reject a superior design for such issues then it's better to comment it directly because reviewers miss these things otherwise.

Reply:
@seanpearsonuk I'm not rejecting this design I had stated the issue that I was facing prior to localize imports. Now I have imported the ansys.fluent.core as per the requirement only instead of importing at top module level import this issue has been resolved and I have updated the code accordingly.

Reply:
@seanpearsonuk @hpohekar `FluentVersion` doesn't support the patch version field. Also, `FluentVersion` is restricted to only few version numbers and won't work generally if this module is reused outside PyFluent.

I think it is better of simply return a tuple from this method.

Reply:
@seanpearsonuk @mkundu1 

Actually I'm trying to make the current scheme eval usage and app utilities changes consistent first. I will address all suggestions in subsequent PRs after merging this PR.

Reply:
https://github.com/ansys/pyfluent/issues/3579

Reply:
> @seanpearsonuk @hpohekar `FluentVersion` doesn't support the patch version field. Also, `FluentVersion` is restricted to only few version numbers and won't work generally if this module is reused outside PyFluent.
> 
> I think it is better of simply return a tuple from this method.
> 
> `f"{response.major}.{response.minor}.{response.patch}"` is also fine and as per the [semvar syntax](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions).

@mkundu1 Ok. Thanks a lot.

Reply:
@seanpearsonuk We will revert this to the original state.